File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
// Package sum of multiples
package summultiples

// Function generate sum of multiples
func SumMultiples(limit int, divisors ...int) int {

	var s int = 0
	var m []int
	for n:=1;n<limit;n++ {
		for _, d:= range divisors {
			//fmt.Println("d=",d)
			if d>0 && n%d == 0 {
				if !in(n,m) {
					m=append(m,n)
					s=s+n
				}
			}
		}
	}
	return s
}

func in(n int, m []int) bool {
	for _,v:= range m {
		if n == v {
			return true
		}
	}
	return false
}

File: solutions/python/sum-of-multiples/2/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
    unique_multiples = set()
    for number in range(1,limit):
        for factor in multiples:
                if number % factor == 0:
                   unique_multiples.add(number)
    sum_of_unique_multiples = sum(unique_multiples)
    return sum_of_unique_multiples
